File: backend/app/schemas/weather.py
"""气象模块 Schema。"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class CurrentWeatherOut(BaseModel):
    time: datetime
    temperature: float | None = None
    feels_like: float | None = None
    humidity: float | None = None
    pressure: float | None = None
    wind_speed: float | None = None
    wind_direction: str | None = None
    weather_code: str | None = None
    weather_desc: str | None = None
    precipitation: float | None = None
    visibility: float | None = None


class ForecastOut(BaseModel):
    time: datetime
    temp_max: float | None = None
    temp_min: float | None = None
    precipitation_sum: float | None = None
    wind_speed_max: float | None = None
    weather_code: str | None = None
    weather_desc: str | None = None


class WeatherHistoryOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    time: datetime
    location_code: str
    temperature: float | None = None
    feels_like: float | None = None
    humidity: float | None = None
    pressure: float | None = None
    wind_speed: float | None = None
    wind_direction: str | None = None
    weather_code: str | None = None
    weather_desc: str | None = None
    precipitation: float | None = None
    visibility: float | None = None
    is_forecast: bool


class SyncResult(BaseModel):
    ok: bool
    location: str | None = None
    temperature: float | None = None
    weather_desc: str | None = None
    alerts: int = 0
    daily: int = 0
    error: str | None = None

File: frontend-user/src/api/weather.ts
import http from './http'

export interface CurrentWeather {
  location_code: string
  temperature: number | null
  feels_like: number | null
  humidity: number | null
  wind_speed: number | null
  wind_direction: string | null
  weather_code: string | null
  weather_desc: string | null
  precipitation: number | null
  visibility: number | null
  time?: string
}

export interface ForecastDay {
  date: string
  time?: string
  temp_max: number | null
  temp_min: number | null
  weather_desc: string | null
  weather_code: string | null
  precipitation: number | null
  is_forecast?: boolean
}

// 获取最新实测天气
export async function getCurrentWeather(location = 'gz'): Promise<CurrentWeather | null> {
  return http.get('/weather/current', { params: { location } })
}

// 获取 7 天预报
export async function getForecast(location = 'gz'): Promise<{ items: ForecastDay[]; total: number }> {
  return http.get('/weather/forecast', { params: { location } })
}

// 获取最近 N 条实测记录（过去的天气）
export async function getHistory(
  limit = 30,
  location = 'gz',
): Promise<{ items: ForecastDay[]; total: number }> {
  return http.get('/weather/history', { params: { location, limit } })
}
